multihost/db-postgresql-with-custom-certificates/test.sh

```diff
@@ -0,0 +1,375 @@
+grant all privileges on schema public to registrar;
+_EOF"
+
+        rlRun "chmod a+x setup.psql"
+        #need set up permission for running postgres
+        rlRun "chmod a+x /tmp/tmp.*"
+        rlFileBackup --clean --missing-ok /var/lib/pgsql /etc/postgresql-setup
+        rlRun "rm -rf /var/lib/pgsql/data"
+        rlRun "postgresql-setup --initdb --unit postgresql"
+        # configure user authentication with md5 and pgsql listening address
+        rlRun "sed -i \"s|^host.*all.*all.*127.0.0.1.*ident|host all all 0.0.0.0/0 md5|\" /var/lib/pgsql/data/pg_hba.conf"
+        rlRun "sed -i \"s|^#listen_addresses.*|listen_addresses = '0.0.0.0'|\" /var/lib/pgsql/data/postgresql.conf"
+        rlServiceStart postgresql
+        sleep 3
+        rlRun "sudo -u postgres psql -f setup.psql"
+        rlRun "sync-set ATTESTATION_SERVER_GENERATE_CERTS_DONE"
+
+        # tenant config
+        rlRun "limeUpdateConf tenant require_ek_cert False"
+        rlRun "limeUpdateConf tenant verifier_ip $ATTESTATION_SERVER_IP"
+        rlRun "limeUpdateConf tenant registrar_ip $ATTESTATION_SERVER_IP"
+        rlRun "limeUpdateConf tenant tls_dir $CERTDIR"
+        rlRun "limeUpdateConf tenant trusted_server_ca '[\"cacert.pem\"]'"
+        rlRun "limeUpdateConf tenant client_cert tenant-cert.pem"
+        rlRun "limeUpdateConf tenant client_key tenant-key.pem"
+
+        #copy cert to agent machine
+        rlRun "scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null $CERTDIR/cacert.pem root@$AGENT_IP:/var/lib/keylime"
+        rlRun "sync-block ANSIBLE_SETUP_DONE ${SETUP_SERVER_IP}" 0 "Waiting for ansible setup of system roles."
+        sleep 5
+        rlRun "limeWaitForVerifier"
+        rlRun "limeWaitForRegistrar"
+
+        #confirm that verifier and registrar accessed and modified databases
+        rlRun -s "sudo -u postgres psql -c 'SELECT datname FROM pg_database;'"
+        rlAssertGrep "verifierdb" $rlRun_LOG
+        rlAssertGrep "registrardb" $rlRun_LOG
+
+        rlRun "sync-set ATTESTATION_SERVER_START"
+        rlRun "sync-block AGENT_STARTS ${AGENT_IP}" 0 "Waiting for ansible setup of system roles."
+        rlRun "limeWaitForAgentRegistration ${AGENT_ID}"
+    rlPhaseEnd
+
+    rlPhaseStartTest "keylime attestation test: Add Agent"
+        # register AGENT and confirm it has passed validation
+        rlRun "cat > script.expect <<_EOF
+set timeout 20
+spawn keylime_tenant -v ${ATTESTATION_SERVER_IP} -t ${AGENT_IP} -u ${AGENT_ID} --runtime-policy /var/tmp/policy.json --include payload-${REVOCATION_SCRIPT_TYPE} --cert default -c add
+expect \"Please enter the password to decrypt your keystore:\"
+send \"keylime\n\"
+expect eof
+_EOF"
+        rlRun "expect script.expect"
+        rlRun "limeWaitForAgentStatus $AGENT_ID 'Get Quote'"
+        rlRun -s "keylime_tenant -c cvlist"
+        rlAssertGrep "{'code': 200, 'status': 'Success', 'results': {'uuids':.*'$AGENT_ID'" $rlRun_LOG -E
+        rlRun "sync-set AGENT_ADD_DONE"
```

Comment:
There could be a delay of several seconds between sync-set and sync-block. And since you do not have sync-block here I am afraid that it won't work reliably. Probably we should more semaphores here, something like
`AGENT_CORRUPTED` and `AGENT_FAILED`.

Reply:
I cannot see this comment in main thread, now I understand what do you mean.